### ISSUE TITLE
per_channel_fp8 pads kv_len to 64 while sm90 kernel asserts pad to CTA_K=128 — direct callers of _qattn_sm90 abort at runtime

### ISSUE THREAD

**Issue by buaasun** [thu-ml/SageAttention#383]
## Summary

`sageattention/quant.py :: per_channel_fp8` pads `kv_len` to a multiple of 64:

```python
padded_len = (kv_len + 63) // 64 * 64
```

But `_qattn_sm90.qk_int8_sv_f8_accum_f32_fuse_v_scale_attn_inst_buf` (and neighbors in `csrc/qattn/qk_int_sv_f8_cuda_sm90.cu`) has `constexpr int CTA_K = 128;` and asserts:

```cpp
assert(value.size(3) >= div_ceil(kv_len, CTA_K) * CTA_K);
```

When `kv_len % 128 ∈ [65, 127]`, `per_channel_fp8`'s pad-to-64 result is 64 short of what the kernel wants and the process aborts with a Fatal Python error.

## Where it bites

`sageattention.core.sageattn_qk_int8_pv_fp8_cuda_sm90` [already works around this by padding V to 128 before calling per_channel_fp8](https://github.com/thu-ml/SageAttention/blob/main/sageattention/core.py#L972-L980), with the acknowledging comment:

```python
# pad v to multiple of 128
# TODO: modify per_channel_fp8 kernel to handle this
kv_len = k.size(seq_dim)
v_pad_len = 128 - (kv_len % 128) if kv_len % 128 != 0 else 0
if v_pad_len > 0:
    v = torch.cat([v, torch.zeros(...)], dim=seq_dim)

v_fp8, v_scale, _ = per_channel_fp8(v, tensor_layout=tensor_layout, smooth_v=False)
```

So users of the top-level `sageattn(...)` API never hit this. But any downstream code that:

1. wants the memory-efficient path (avoid the `torch.cat` allocation on the full V) and/or
2. wants finer control over quantization / K-mean smoothing / kernel dispatch

typically calls `per_channel_fp8` directly and then `_qattn_sm90.*` directly, forgetting the pad-128 step. Example downstream: `MiniMaxH3MemoryEfficientSageAttentionPatch` in `kijai/ComfyUI-KJNodes`, which crashes on MiniMax H3 with any long prompt (PR filed separately at kijai/ComfyUI-KJNodes#729).

## Reproduction

- SM90 GPU (tested on H20)
- Any `kv_len` such that `kv_len % 128 ∈ [65, 127]` (65, 66, ..., 127 all crash; 0-64 and 128 are fine because `per_channel_fp8`'s pad-64 result matches or exceeds pad-128)
- Direct call to `per_channel_fp8` followed by `_qattn_sm90.qk_int8_sv_f8_accum_f32_fuse_v_scale_attn_inst_buf`

Sketched minimal repro:

```python
import torch
from sageattention.quant import per_channel_fp8
from sageattention import _qattn_sm90
b, seq, h, d = 1, 15529, 32, 128   # 15529 % 128 = 89 → in failing range
v = torch.randn(b, seq, h, d, device='cuda', dtype=torch.bfloat16)
v_fp8, v_scale, _ = per_channel_fp8(v, tensor_layout="NHD", smooth_v=False)
# v_fp8.size(3) == 15552 (padded to 64); kernel wants >= 15616 (padded to 128)
# calling the kernel with v_fp8 aborts the process
```

## Suggested fix

Two options:

**Option 1 — change `per_channel_fp8` pad to `CTA_K = 128` unconditionally**:

```python
padded_len = (kv_len + 127) // 128 * 128
```

Requires checking that `_fused.transpose_pad_permute_cuda` and `_fused.mean_scale_fuse_quant_cuda` don't hardcode a 64 pad internally. If they take `padded_len` purely as an argument this is a one-line change with no ABI break.

**Option 2 — accept an explicit `pad_to` argument** (more surgical):

```python
def per_channel_fp8(v, tensor_layout="HND", scale_max=448.0, smooth_v=True, pad_to=64):
    ...
    padded_len = (kv_len + pad_to - 1) // pad_to * pad_to
```

Callers pick 128 when they'll hand the result to the sm90 kernel. This matches the intent of the `# TODO` in `core.py` and won't regress anyone who is fine with pad-64.

Happy to send a PR for either option — please indicate a preference.
